### PR TITLE
add prometheus for event/eventers

### PR DESCRIPTION
**Pull request by huangyuqi**
add prometheus for event/eventers
- last event timestamp
- last housekeep timestamp
- total number of events
- time spent for scraping events
- time spent for writing events (total and on per sink basis)

### CODE REVIEW

**Comment by huangyuqi** [comment]
@jimmidyson Please take a look when you get a chance. Thanks so much.
And, I  don't understand the meaning of " number of runs in the current houskeep interval", so i leave this to another commit. Please @mwielgus help to explain for me, thanks.

**Comment by jimmidyson** [comment]
Travis compile error:

```
# k8s.io/heapster/events/sources/kubernetes
events/sources/kubernetes/kubernetes_source.go:101: invalid argument result (type core.EventBatch) for len
```

File: events/manager/manager.go [hunk 1]
```diff
@@ -17,11 +17,26 @@ package manager
...
+)
 
-	"github.com/golang/glog"
+var (
+	// Last time of eventer housekeep since unix epoch in seconds
+	lastHousekeepTimestamp = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: "heapster",
+			Subsystem: "eventer",
+			Name:      "last_time_seconds",
```
Comment (jimmidyson) [inline events/manager/manager.go]: `last_housekeep_seconds`?

File: events/manager/manager.go [hunk 2]
```diff
@@ -17,11 +17,26 @@ package manager
...
+	"github.com/prometheus/client_golang/prometheus"
 	"k8s.io/heapster/events/core"
+)
 
-	"github.com/golang/glog"
+var (
+	// Last time of eventer housekeep since unix epoch in seconds
+	lastHousekeepTimestamp = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: "heapster",
```
Comment (jimmidyson) [inline events/manager/manager.go]: This should be `eventer`.

File: events/sinks/manager.go [hunk 3]
```diff
@@ -27,6 +28,23 @@ const (
 	DefaultSinkStopTimeout         = 60 * time.Second
 )
 
+var (
+	// Time spent exporting events to sink in microseconds.
+	exporterDuration = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Namespace: "heapster",
```
Comment (jimmidyson) [inline events/sinks/manager.go]: `eventer`

File: events/sources/kubernetes/kubernetes_source.go [hunk 4]
```diff
@@ -36,6 +37,37 @@ const (
 	LocalEventsBufferSize = 100000
 )
 
+var (
+	// Last time of event since unix epoch in seconds
+	lastEventTimestamp = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: "heapster",
```
Comment (jimmidyson) [inline events/sources/kubernetes/kubernetes_source.go]: `eventer`

File: events/sources/kubernetes/kubernetes_source.go [hunk 5]
```diff
@@ -36,6 +37,37 @@ const (
...
+			Subsystem: "events",
+			Name:      "last_time_seconds",
+			Help:      "Last time of event since unix epoch in seconds.",
+		})
+	totalEventsNum = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Namespace: "heapster",
```
Comment (jimmidyson) [inline events/sources/kubernetes/kubernetes_source.go]: `eventer`

File: events/sources/kubernetes/kubernetes_source.go [hunk 6]
```diff
@@ -36,6 +37,37 @@ const (
...
+			Subsystem: "events",
+			Name:      "events_total_number",
+			Help:      "The total number of events.",
+		})
+	scrapEventsDuration = prometheus.NewSummary(
+		prometheus.SummaryOpts{
+			Namespace: "heapster",
```
Comment (jimmidyson) [inline events/sources/kubernetes/kubernetes_source.go]: `eventer`

File: events/sources/kubernetes/kubernetes_source.go [hunk 7]
```diff
@@ -36,6 +37,37 @@ const (
...
+			Subsystem: "events",
```
Comment (jimmidyson) [inline events/sources/kubernetes/kubernetes_source.go]: `scraper`

File: events/sources/kubernetes/kubernetes_source.go [hunk 8]
```diff
@@ -36,6 +37,37 @@ const (
 	LocalEventsBufferSize = 100000
 )
 
+var (
+	// Last time of event since unix epoch in seconds
+	lastEventTimestamp = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: "heapster",
+			Subsystem: "events",
```
Comment (jimmidyson) [inline events/sources/kubernetes/kubernetes_source.go]: `scraper`

File: events/sources/kubernetes/kubernetes_source.go [hunk 9]
```diff
@@ -36,6 +37,37 @@ const (
...
+			Name:      "last_time_seconds",
+			Help:      "Last time of event since unix epoch in seconds.",
+		})
+	totalEventsNum = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Namespace: "heapster",
+			Subsystem: "events",
```
Comment (jimmidyson) [inline events/sources/kubernetes/kubernetes_source.go]: `scraper`

File: events/manager/manager.go [hunk 10]
```diff
@@ -17,11 +17,26 @@ package manager
...
+			Subsystem: "eventer",
```
Comment (jimmidyson) [inline events/manager/manager.go]: `manager`?

File: events/manager/manager.go (hunk 1 above)
Comment (jimmidyson) [inline events/manager/manager.go]: No keep as is.

**Comment by jimmidyson** [comment]
In general: namespace should be the app (in this case `eventer`) & Subsystem should be the area of functionality (`exporter`, `scraper`, etc). Scraper might have a better name in this case tbh. Perhaps we should change to `sources`, `sinks`? & do the same with heapster metrics.

**Comment by huangyuqi** [comment]
I got it. Thanks @jimmidyson .
Let me check the CI issue first.

**Comment by piosz** [comment]
@k8s-bot ok to test

File: events/sources/kubernetes/kubernetes_source.go [hunk 11]
```diff
@@ -36,6 +37,37 @@ const (
...
+			Help:      "Last time of event since unix epoch in seconds.",
+		})
+	totalEventsNum = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Namespace: "eventer",
+			Subsystem: "scraper",
+			Name:      "events_total_number",
+			Help:      "The total number of events.",
+		})
+	scrapEventsDuration = prometheus.NewSummary(
```
Comment (jimmidyson) [inline events/sources/kubernetes/kubernetes_source.go]: nit: `scrapeEventsDuration`

File: events/sources/kubernetes/kubernetes_source.go [hunk 12]
```diff
@@ -36,6 +37,37 @@ const (
...
+			Namespace: "eventer",
+			Subsystem: "scraper",
+			Name:      "last_time_seconds",
+			Help:      "Last time of event since unix epoch in seconds.",
+		})
+	totalEventsNum = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Namespace: "eventer",
+			Subsystem: "scraper",
+			Name:      "events_total_number",
```
Comment (jimmidyson) [inline events/sources/kubernetes/kubernetes_source.go]: Can drop `_number` - these are all numbers :)

**Comment by mwielgus** [comment]
LGTM

We will merge the PR as soon as we switch Kubernetes HEAD to use new Heapster (alpha). Right now we are fixing bugs and don't want to introduce any new code.

**Pull request merged**